test/bandit/http1/request_test.exs

```diff
@@ -108,6 +108,64 @@ defmodule HTTP1RequestTest do
       assert Jason.decode!(body)["port"] == 1234
     end
 
+    test "derives port from host header for all valid ports", context do
+      client = SimpleHTTP1Client.tcp_client(context)
+
+      for port <- 0..65_535 do
```

Comment:
This is a lot of requests! We could probably stand to do a subset of this (either random, some set of pre-selected ones, or using a `//step` on the range

Reply:
I kind of thought that's what you intended when you wanted my other tests ported to these modules, but I pushed a more sane version.